File: models/contact_model.go
package models

// Contact 通讯录
type Contact struct {
	ID              int64  `orm:"auto;pk;column(id)" json:"id"`
	FromAccount     int64  `orm:"type(bigint);column(from_account)" json:"from_account"`
	ToAccount       int64  `orm:"type(bigint);column(to_account)" json:"to_account"`
	IsSessionEnd    int    `orm:"default(0),column(is_session_end)" json:"is_session_end"` // 1 已结束对话 0 未结束对话
	LastMessage     string `orm:"null;type(text);column(last_message)" json:"last_message"`
	Delete          int    `orm:"default(0);column(delete)" json:"delete"` // 1 已删除 0 未删除
	LastMessageType string `orm:"null;type(text);column(last_message_type)" json:"last_message_type"`
	CreateAt        int64  `orm:"type(bigint);column(create_at)" json:"create_at"`
}

File: ui/kefu_workbench/lib/models/contact_model.dart
class ContactModel {
  int id;
  int cid;
  int fromAccount;
  int toAccount;
  String lastMessage;
  int isSessionEnd;
  String lastMessageType;
  int uid;
  String avatar;
  String address;
  String nickname;
  String phone;
  int platform;
  int online;
  int read;
  int updateAt;
  String remarks;
  int lastActivity;
  int createAt;
  int contactCreateAt;

  ContactModel(
      {this.id,
      this.cid,
      this.fromAccount,
      this.toAccount,
      this.lastMessage,
      this.isSessionEnd,
      this.lastMessageType,
      this.uid,
      this.avatar,
      this.address,
      this.nickname,
      this.phone,
      this.platform,
      this.online,
      this.read,
      this.updateAt,
      this.remarks,
      this.lastActivity,
      this.createAt,
      this.contactCreateAt});

  ContactModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    cid = json['cid'];
    fromAccount = json['from_account'];
    toAccount = json['to_account'];
    lastMessage = json['last_message'];
    isSessionEnd = json['is_session_end'];
    lastMessageType = json['last_message_type'];
    uid = json['uid'];
    avatar = json['avatar'];
    address = json['address'];
    nickname = json['nickname'];
    phone = json['phone'];
    platform = json['platform'];
    online = json['online'];
    read = json['read'];
    updateAt = json['update_at'];
    remarks = json['remarks'];
    lastActivity = json['last_activity'];
    createAt = json['create_at'];
    contactCreateAt = json['contact_create_at'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['cid'] = this.cid;
    data['from_account'] = this.fromAccount;
    data['to_account'] = this.toAccount;
    data['last_message'] = this.lastMessage;
    data['is_session_end'] = this.isSessionEnd;
    data['last_message_type'] = this.lastMessageType;
    data['uid'] = this.uid;
    data['avatar'] = this.avatar;
    data['address'] = this.address;
    data['nickname'] = this.nickname;
    data['phone'] = this.phone;
    data['platform'] = this.platform;
    data['online'] = this.online;
    data['read'] = this.read;
    data['update_at'] = this.updateAt;
    data['remarks'] = this.remarks;
    data['last_activity'] = this.lastActivity;
    data['create_at'] = this.createAt;
    data['contact_create_at'] = this.contactCreateAt;
    return data;
  }
}